src/basic/log.c

```diff
@@ -1584,11 +1585,21 @@ LogContext* log_context_detach(LogContext *c) {
 }
 
 LogContext* log_context_new(char **fields, bool owned) {
+        if (!fields)
+                return NULL;
+
+        LIST_FOREACH(ll, i, _log_context)
+                if (i->fields == fields) {
```

Comment:
I think it is, especially since it might avoid us adding the same N fields to a log message multiple times. Also, `n` is super small here, so I don't think the complexity matters that much.

Reply:
Also, we check against the same exact pointer to avoid making the check too costly. The idea is to allow pushing the same fields (same pointer) on the log context many times and not having to worry about affecting performance. This allows us to be much more liberal in adding `LOG_CONTEXT_PUSH()` calls everywhere as long as we use the same fields pointer every time.